pkg/operator/ceph/cluster/mon/node.go

```diff
@@ -52,14 +52,14 @@ func getNodeInfoFromNode(n v1.Node) (*NodeInfo, error) {
 	}
 
 	for _, ip := range n.Status.Addresses {
-		if ip.Type == v1.NodeExternalIP || ip.Type == v1.NodeInternalIP {
-			logger.Debugf("using IP %s for node %s", ip.Address, n.Name)
+		if ip.Type == v1.NodeInternalIP {
```

Comment:
Do we need an option in the CR so they can select the external or internal? In some clusters they may need the external ip if they are intending to make it available as an external cluster.

Reply:
I'd say we wait until the request comes in? I think most of the users have external = internal, that's why no one never noticed.

I think it's fine sticking with internal and then maybe open it up if someone wants to expose the cluster through external IPs.

Reply:
The current behavior would certainly be unpredictable anyway if the external and internal were different, so the proposed behavior is better. But I also wonder if the external ip will ever be set when the internal ip is not? In that case we could pick the external ip.

Reply:
The thing is it's not working when we pick up the external IP as shown in #5495 so that seems risky to me.

Reply:
My question is if it is possible that the internal IP won't be set, even though the external IP is set. In that case, we wouldn't bind to either one. But that doesn't seem likely so let's go with this for now.